server-common/src/main/java/org/apache/kafka/server/config/KafkaLogConfigs.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.server.config;
+
+import org.apache.kafka.common.config.TopicConfig;
+
+import static org.apache.kafka.server.common.MetadataVersion.IBP_3_0_IV1;
+import static org.apache.kafka.server.config.ServerTopicConfigSynonyms.LOG_PREFIX;
+
+/**
+ * Common home for broker-side log configs which need to be accessible from the libraries shared
+ * between the broker and the multiple modules in Kafka.
+ *
+ * Note this is an internal API and subject to change without notice.
+ */
+public class KafkaLogConfigs {
```

Comment:
Do we need to add prefix "kafka" here? sorry that I try to figure out the naming rule. It seems to me the prefix "kafka" means the class is used to collect other "xxxConfigs" constants and to be the central place for kafka xxx-related configs. For example, `KafkaSecurityConfigs` collects all security-related configs from `BrokerSecurityConfigs`, `SaslConfigs`, `SslConfigs`, etc

Reply:
I can rename it to `ServerLogConfigs` or `BrokerLogConfigs` but we already have `LogConfig`

Reply:
Just pushed an update. The problem right now that we have the name of the config type without any unnecessary prefixes has been preserved in other modules and am trying to avoid making this pr bigger than it should be. I think at some point we need to revisit some of the existing Config files as some of them breaks few rules like using `PROP` instead of `CONFIG` suffix and use `DEFAULT` as prefix instead of suffix and some of the classes is `Config` instead of `Configs` as well.

Reply:
> I think at some point we need to revisit some of the existing Config files as some of them breaks few rules like using PROP instead of CONFIG suffix and use DEFAULT as prefix instead of suffix and some of the classes is Config instead of Configs as well.

agree!